openwisp_radius/api/views.py

```diff
@@ -210,20 +212,22 @@ def post(self, request, *args, **kwargs):
         serializer.is_valid(raise_exception=True)
         username = serializer.validated_data.get('username')
         password = serializer.validated_data.get('password')
-        user = self.get_user(request, username)
+        user = self.get_user(request, username, password)
         if user and self.authenticate_user(request, user, password):
             return Response(self.accept_attributes, status=self.accept_status)
         if app_settings.API_AUTHORIZE_REJECT:
             return Response(self.reject_attributes, status=self.reject_status)
         else:
             return Response(None, status=200)
 
-    def get_user(self, request, username):
+    def get_user(self, request, username, password):
         """
         return active user or ``None``
         """
         try:
-            user = User.objects.get(username=username, is_active=True)
+            user = auth_backend.authenticate(
+                request, username, password
+            ) or User.objects.get(username=username, is_active=True)
```

Comment:
You can't run an authentication in this method, which is supposed to simply fetch the user.
It's very wrong for multiple reasons.

It breaks the radius user token authentication mechanism, because it checks the password, but end users authenticate using one time tokens (see `RadiusToken`).

The check password or check radius token operation is performed by this method:

https://github.com/openwisp/openwisp-radius/blob/5752ba2a419126722d637bc42edf4aa4a2296992/openwisp_radius/api/views.py#L244-L253

If you want to reuse the logic to fetch the user from `UserAuthenticationBackend`, which sounds good because it allows us to avoid the repetition, please separate the logic that fetches the user in the authentication backend, create a `get_user` method (which doens't check the password) so it can be reused here.